profiling-ffi/src/exporter.rs

```diff
@@ -169,6 +170,28 @@ unsafe fn into_vec_files<'a>(slice: Slice<'a, File>) -> Vec<exporter::File<'a>>
         .collect()
 }
 
+#[repr(C)]
+pub enum ExporterRequestBuildResult {
+    Ok(Box<Request>),
+    Err(ddcommon_ffi::Vec<u8>),
+}
+
+#[cfg(test)]
+impl From<ExporterRequestBuildResult> for Result<Box<Request>, String> {
+    fn from(result: ExporterRequestBuildResult) -> Self {
+        match result {
+            ExporterRequestBuildResult::Ok(ok) => Ok(ok),
+            ExporterRequestBuildResult::Err(err) => {
+                // Safety: not generally safe but this is for test code.
+                Err(unsafe { String::from_utf8_unchecked(err.into()) })
```

Comment:
Minor: I'm curious, why is this unsafe? Because the error may not be representable in utf8?

Reply:
Yes, and since it could come from C code, there's no guarantee.

Reply:
Aaah, I see. In our particular use, we know this is never going to happen (because we're not going to take in a `ExporterRequestBuildResult` and then look at the contents of its `Err`), but I guess the type system can't know that.

Reply:
The `Error` type I added would make this safer, at least if people follow the API and comments :)